src/cp/slipharden.cxx

```diff
@@ -937,6 +937,302 @@ void SimpleLinearHardening::consistency(Lattice & L) const
   }
 }
 
+LANLTiModel::LANLTiModel(ParameterSet & params):
+    SlipHardening(params),
+    tau_0_(params.get_object_parameter_vector<Interpolate>("tau_0")),
+    C_st_(params.get_object_parameter<SquareMatrix>("C_st")),
+    mu_(params.get_object_parameter_vector<Interpolate>("mu")),
+	  k1_(params.get_object_parameter_vector<Interpolate>("k1")), 
+    k2_(params.get_object_parameter_vector<Interpolate>("k2")),
+    X_s_(params.get_parameter<double>("X_s")),
+    varprefix_(params.get_parameter<std::string>("varprefix")), 
+    twinprefix_(params.get_parameter<std::string>("twinprefix"))
+{ 
+  // Shouldn't tau_0 be 24 and C_st be 12?
+  // Long term you should fix the /2
+  if (C_st_->n() != tau_0_.size()/2) {
```

Comment:
You need to fix all of these (here and below).

So you need to add a few checks here.

`k1_` and `k2_` should have the same size, equal to the number of slip systems.

`C_st_` should have `m()` = number of slip systems and n() = number of twin systems

`tau_0_` should have size equal to the number of twin systems.

I suggest you check the sizes of all these things here and then store two `size_t` parameters in the class: `nslip_` and `ntwin_` and use them in your loops below.